base/traits.jl

```diff
@@ -1,17 +1,47 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 ## numeric/object traits
-# trait for objects that have an ordering
+
+"""
+    OrderedStyle()
+ 
+An abstract type of a trait that signals if some type `T` guarantees
+that a call `isless(::T, ::T)` will return `true` or `false`.
+
+It has two concrete subtypes [`Ordered`](@ref) and
+[`Unordered`](@ref), which is the default.
+
+Types supporting [`isless`](@ref) are recommended to support
+[`Ordered`](@ref) trait to enable optimizations.
```

Comment:
Other existing traits document the function rather than the abstract type in their docstring. Maybe simpler copy the structure used for `IteratorEltype`:
```suggestion
    OrderStyle(T::Type) -> OrderStyle

Given a type `T`, return one of the following values:
- [`Ordered()`](@ref) if type `T` guarantees that a call `isless(::T, ::T)`
will return `true` or `false`.
- [`Unordered()`](@ref) otherwise (this is the default).

Types supporting [`isless`](@ref) are recommended to return
[`Ordered()`](@ref) to enable optimizations.
```